src/backends/locked.ts

```diff
@@ -47,7 +51,8 @@ export class LockedFS<FS extends FileSystem> implements FileSystem {
 	public async lock(path: string): Promise<MutexLock> {
 		if (this.locks.has(path)) {
 			// Non-null assertion: we already checked locks has path
-			await this.locks.get(path)!.promise;
```

Comment:
1. This is not a promise. `await`ing the lock has no effect and is a bug.
2. Please inline: `await this.locks.get(path)!.at(-1).promise`